codegenerator/cli/templates/dynamic/codegen/src/TestHelpers.res.hbs

```diff
@@ -168,31 +169,33 @@ module EventFunctions = {
   }
 
   module MockBlock = {
+    @genType
     type t = {
-      {{#each field_selection.block_fields as | field |}}
+      {{#each aggregated_field_selection.block_fields as | field |}}
       {{field.name.camel}}?: {{field.res_type}},
       {{/each}}
     }
```

Comment:
Alternatively we could generate the partial type per event, but I got tired here, so I thought using aggregated fields for the mock options will work pretty well.

Reply:
Hmmm, I think we probably should otherwise the test might not help a user say if they use js and try and access an unavailable field in a handler.

I don't think it should block getting this in though.

Reply:
Oh no, I misunderstood actually, the test would still reveal this. So I think it's totally fine for the mocker to provide all the aggregated fields.